adlfs/spec.py

```diff
@@ -73,6 +73,12 @@
 _SOCKET_TIMEOUT_DEFAULT = object()
 
 
+def _get_user_agent():
```

Comment:
In order to break the circular loop here, I think it would be worth moving this logic: https://github.com/anjaliratnam-msft/adlfs/blob/f64dff2b2ce62dd2d194e0be02804104b4505d7d/adlfs/__init__.py#L6-L11 to the `utils.py` module and then have the `__init__.py` import `__version__` and `version_tuple` and this module can import `__version__`. Then we can go back to having a module level `_USER_AGENT` constant.

Generally, a `utils.py` module will have all of the shared code and not have any intra project imports which is usually a safe approach to avoiding import errors.